File: core/src/test/java/io/github/herrromich/batch/job/TestTask.java
package io.github.herrromich.batch.job;

import io.github.herrromich.batch.FailLevel;
import io.github.herrromich.batch.Task;
import io.github.herrromich.batch.TaskContext;
import io.github.herrromich.batch.TaskPriorities;
import org.jetbrains.annotations.NotNull;

import java.util.Set;

public abstract class TestTask implements Task {
    @Override
    public int getPriority() {
        return TaskPriorities.DEFAULT();
    }

    @NotNull
    @Override
    public FailLevel getFailLevel() {
        return FailLevel.DEFAULT();
    }

    @NotNull
    @Override
    public Set<String> getConsumables() {
        return Set.of();
    }

    @NotNull
    @Override
    public Set<String> getProducibles() {
        return Set.of();
    }

    @Override
    public void execute(@NotNull TaskContext context) {
        
    }
}

File: core/src/testFixtures/kotlin/io/github/herrromich/batch/TestTask.kt
package io.github.herrromich.batch

import mu.KotlinLogging

private val logger = KotlinLogging.logger { }

abstract class TestTask(override val name: String) : Task {
    override val priority: Int
        get() = TaskPriorities.DEFAULT
    override val failLevel: FailLevel
        get() = FailLevel.DEFAULT
    override val consumables: Set<String>
        get() = emptySet()
    override val producibles: Set<String>
        get() = emptySet()
    override fun execute(context: TaskContext) {
        logger.info { "Executiong test task: \"${name}\"" }
    }
}
